### PR TITLE
Add New lr scheduler

### PR DESCRIPTION
**Pull request by sdbds**
Change Lr schedulers library from diffusers to transformers, they use same library but transformers had more lr schedulers.
Considering the community has always wanted custom learning rate, I think these newly added regulators will meet the demand.

[Add inverse sqrt learning rate scheduler](https://github.com/huggingface/transformers/pull/21495)
image
new argument lr_scheduler_timescale,default to warms_up_steps

[Add cosine with min lr scheduler](https://github.com/huggingface/transformers/pull/29341)
When set the num_steps=100, num_warmup_steps=10, lr=0.2, min_lr=0.01. The learning rate looks like:
image

new argument lr_scheduler_min_lr_ratio,default to 0 as cosine lr scheduler.

[Add WSD scheduler](https://github.com/huggingface/transformers/pull/30231)
The ladder scheduler that so many people have been wanting.
image

new argument lr_decay_steps,
new argument lr_scheduler_min_lr_ratio,default to 0.

need update requirement transformers==4.41.2

### CODE REVIEW

**Comment by sdbds** [comment]
Fix bugs now, i think it works well.
@kohya-ss

**Comment by sdbds** [comment]
> Thank you! However, removing PIECEWISE_CONSTANT will likely impact users. I would like to update the code after merging so that it can be used, but it will take some time. I'd like to prioritize some other PRs. If you could fix this, that would be great.

OK，just keep diffusers import for backup using, use 
```
name = SchedulerType(name) or DiffusersSchedulerType(name)
schedule_func = TYPE_TO_SCHEDULER_FUNCTION[name] or DIFFUSERS_TYPE_TO_SCHEDULER_FUNCTION[name]
```
to judge if using PIECEWISE_CONSTANT.
And i add parser type for input float with warmup and decay ratio, don't need to calculate training total steps.

**Comment by sdbds** [comment]
> Sorry for bothering you again. Is there a reason to upgrade the version of the library other than transformers? It requires some more comprehensive testing, which takes time.

The main thing is to upgrade the transformer version, and after upgrading the transformer version the other two dependencies will also ask for an update, so it's three version updates.
Since the PR is older, I reviewed the latest three dependency update histories, and there don't seem to be any major bug fixes as well as disruptive changes.

File: library/train_util.py [hunk 1]
```diff
@@ -4331,13 +4368,13 @@ def wrap_check_needless_num_warmup_steps(return_vals):
         # logger.info(f"adafactor scheduler init lr {initial_lr}")
         return wrap_check_needless_num_warmup_steps(transformers.optimization.AdafactorSchedule(optimizer, initial_lr))
 
-    name = SchedulerType(name)
-    schedule_func = TYPE_TO_SCHEDULER_FUNCTION[name]
+    name = SchedulerType(name) or DiffusersSchedulerType(name)
```
[the remark this answers is not in the document]
Comment (sdbds) [reply library/train_util.py:4371]: I forgot about ValueError, I guess it can be handled separately.
